src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,986 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/ledger/LedgerMaster.h>
+#include <ripple/app/ledger/LedgerToJson.h>
+#include <ripple/beast/utility/temp_dir.h>
+#include <ripple/core/ConfigSections.h>
+#include <ripple/nodestore/DatabaseShard.h>
+#include <ripple/nodestore/DummyScheduler.h>
+#include <ripple/nodestore/Manager.h>
+#include <ripple/nodestore/impl/DecodedBlob.h>
+#include <ripple/nodestore/impl/EncodedBlob.h>
+#include <ripple/nodestore/impl/Shard.h>
+#include <test/jtx.h>
+#include <test/nodestore/TestBase.h>
+
+namespace ripple {
+namespace NodeStore {
+
+// Tests DatabaseShard class
+//
+class DatabaseShard_test : public TestBase
+{
+    static constexpr std::uint32_t maxSizeGb = 10;
+    static constexpr std::uint32_t ledgersPerShard = 256;
+    static constexpr std::uint32_t earliestSeq = ledgersPerShard + 1;
+    static constexpr std::uint32_t dataSizeMax = 4;
+    static constexpr std::uint32_t iniAmount = 1000000;
+    static constexpr std::uint32_t nTestShards = 4;
+    static constexpr std::uint32_t shardStoreTimeout = 60;
```

Comment:
Consider using the std::chrono timing API: https://github.com/HowardHinnant/rippled/commit/d43f1682065b8ace5d5d8360ed6609b4e69b7682

Reply:
Done